business_objects/monitor.py

```diff
@@ -0,0 +1,424 @@
+from typing import Any, List
+from . import general
+from .. import enums
+
+
+def get_all_tasks(project_id: str = None, only_running: bool = False) -> List[Any]:
+    query = f""" 
+    SELECT tasks.*, orga.name organization_name
+    FROM (
+        ({__select_running_information_source_payloads(project_id, only_running)})
+        UNION
+        ({__select_running_attribute_calculation_tasks(project_id, only_running)})
+        UNION
+        ({__select_running_tokenization_tasks(project_id, only_running)})
+        UNION
+        ({__select_running_embedding_tasks(project_id, only_running)})
+        UNION
+        ({__select_running_weak_supervision_tasks(project_id, only_running)})
+        UNION
+        ({__select_running_upload_tasks(project_id, only_running)})
+    ) tasks
+    JOIN project p
+    ON p.id = tasks.project_id
+    JOIN organization orga
+    ON orga.id = p.organization_id
+    """
+
+    return general.execute_all(query)
+
+
+def cancel_all_running_tasks(project_id: str = None):
+    set_information_source_payloads_to_failed(project_id)
+    set_attribute_calculation_to_failed(project_id)
+    set_embedding_to_failed(project_id)
+    set_record_tokenization_task_to_failed(project_id)
+    set_upload_task_to_failed(project_id)
+    set_weak_supervision_to_failed(project_id)
+    general.commit()
+
+
+def set_information_source_payloads_to_failed(
+    project_id: str = None, payload_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE information_source_payload
+    SET state = '{enums.PayloadState.FAILED.value}'
+    WHERE state = '{enums.PayloadState.STARTED.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+
+    if payload_id:
+        query = (
+            query
+            + f"""
+        AND id = '{payload_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_attribute_calculation_to_failed(
+    project_id: str = None, attribute_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE attribute
+    SET state = '{enums.AttributeState.FAILED.value}'
+    WHERE state = '{enums.AttributeState.RUNNING.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if attribute_id:
+        query = (
+            query
+            + f"""
+        AND id = '{attribute_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_record_tokenization_task_to_failed(
+    project_id: str = None, task_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE record_tokenization_task
+    SET state = '{enums.TokenizerTask.STATE_FAILED.value}'
+    WHERE state = '{enums.TokenizerTask.STATE_IN_PROGRESS.value}'
```

Comment:
both CREATED and IN_PROGRESS since even the created ones should break if we restart.

Reply:
Done